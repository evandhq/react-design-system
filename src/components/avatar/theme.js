import defaultTheme from '../themes';

const avatarTheme = {
  borderRadius: defaultTheme.borderRadius.low,
  size: {
    lg: 128,
    md: 64,
    sm: 32,
  },
};

export default avatarTheme;
