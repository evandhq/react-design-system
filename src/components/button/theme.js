import defaultTheme from '../themes';

const buttonTheme = {
  size: {
    lg: {
      padding: '4px 48px',
      font: 20,
      minWidth: 172,
      height: 56,
      margin: 16,
    },
    md: {
      padding: '4px 32px',
      font: 14,
      minWidth: 128,
      height: 48,
      margin: 12,
    },
    sm: {
      padding: '2px 16px',
      font: 10,
      minWidth: 80,
      height: 36,
      margin: 8,
    },
  },
  colors: defaultTheme.colors,
  borderRadius: defaultTheme.borderRadius.low,
};

export default buttonTheme;
