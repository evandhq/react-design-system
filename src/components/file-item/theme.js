import defaultTheme from '../themes';

const theme = {
  colors: {
    riverBedDark: defaultTheme.colors.riverBedDark,
    riverBed: defaultTheme.colors.riverBed,
  },
  borderRadius: defaultTheme.borderRadius.high,
  spacing: defaultTheme.spacing,
};

export default theme;
