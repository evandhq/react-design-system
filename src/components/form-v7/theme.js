import defaultTheme from '../themes';

const theme = {
  borderRadius: defaultTheme.borderRadius.high,
  borderColor: {
    normal: defaultTheme.colors.gray600,
    hover: defaultTheme.colors.blue400,
    focus: defaultTheme.colors.blue600,
  },
  disabled: {
    background: defaultTheme.colors.gray200,
    color: defaultTheme.colors.gray400,
  },
  colors: {
    red: defaultTheme.colors.red600,
    blue: defaultTheme.colors.blue600
  }
};

export default theme;
