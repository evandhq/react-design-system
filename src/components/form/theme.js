import defaultTheme from '../themes';

const theme = {
  borderRadius: defaultTheme.borderRadius.low,
  borderColor: {
    normal: defaultTheme.colors.gray800,
    hover: defaultTheme.colors.blue400,
    focus: defaultTheme.colors.blue600,
  },
  disabled: {
    background: defaultTheme.colors.gray200,
    color: defaultTheme.colors.gray400,
  },
};

export default theme;
