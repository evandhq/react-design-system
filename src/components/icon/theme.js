import defaultTheme from '../themes';

const iconTheme = {
  size: {
    lg: 24,
    md: 20,
    sm: 16,
  },
  marginSize: {
    lg: 12,
    md: 10,
    sm: 8,
  },
  colors: defaultTheme.colors,
};

export default iconTheme;
