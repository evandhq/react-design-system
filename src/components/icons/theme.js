import defaultTheme from '../themes';

const theme = {
  color: defaultTheme.colors.riverBedDark,
  sizes: {
    large: 24,
    largeMargin: 12,
    medium: 20,
    mediumMargin: 10,
    small: 16,
    smallMargin: 8,
  },
};

export default theme;
