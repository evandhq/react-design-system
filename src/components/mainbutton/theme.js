import defaultTheme from '../themes';

const buttonTheme = {
  size: {
    lg: {
      padding: '10px 48px',
      font: 20,
      minWidth: 172,
    },
    md: {
      padding: '10px 32px',
      font: 14,
      minWidth: 128,
    },
    sm: {
      padding: '8px 16px',
      font: 10,
      minWidth: 80,
    },
  },
  colors: defaultTheme.colors,
};

export default buttonTheme;
