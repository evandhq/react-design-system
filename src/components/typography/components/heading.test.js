import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components';


import Heading from './heading';
import { findByTestAtrr } from '../../helpers';
import theme from '../theme';

const setupShallow = (props = {}) => (
  shallow(<Heading level={props.level} {...props}>{props.children}</Heading>)
);

const setupMount = (props = {}) => (
  mount(<Heading level={props.level} {...props}>{props.children}</Heading>)
);

describe('Heading component', () => {
  it('Should render large H1 component without error', () => {
    const component = setupShallow({ children: 'تست', level: 1 });
    const H1 = findByTestAtrr(component, 'h1-tag');
    expect(H1.length).toBe(1);
  });

  it('Should render large H1 component without error', () => {
    const H1 = setupMount({ children: 'تست', level: 1 });
    expect(H1).toHaveStyleRule('font-size', `${theme.h1SizeLarge}px`);
    expect(H1).toHaveStyleRule('line-height', `${theme.h1SizeLarge * theme.lineHeightRatio}px`);
    expect(H1).toHaveStyleRule('color', theme.defaultColor);
    H1.unmount();
  });

  it('Should render H2 component without error', () => {
    const component = setupShallow({ children: 'تست', level: 2 });
    const H2 = findByTestAtrr(component, 'h2-tag');
    expect(H2.length).toBe(1);
  });

  it('Should render small H2 component without error', () => {
    const H2 = setupMount({ children: 'تست', level: 2, size: 'sm' });
    expect(H2).toHaveStyleRule('font-size', `${theme.h2SizeSmall}px`);
    expect(H2).toHaveStyleRule('line-height', `${theme.h2SizeSmall * theme.lineHeightRatio}px`);
    expect(H2).toHaveStyleRule('color', theme.defaultColor);
    H2.unmount();
  });

  it('Should render H3 component without error', () => {
    const component = setupShallow({ children: 'تست', level: 3 });
    const H3 = findByTestAtrr(component, 'h3-tag');
    expect(H3.length).toBe(1);
  });

  it('Should render large H3 component without error', () => {
    const H3 = setupMount({ children: 'تست', level: 3, size: 'lg' });
    expect(H3).toHaveStyleRule('font-size', `${theme.h3SizeLarge}px`);
    expect(H3).toHaveStyleRule('line-height', `${theme.h3SizeLarge * theme.lineHeightRatio}px`);
    expect(H3).toHaveStyleRule('color', theme.defaultColor);
    H3.unmount();
  });

  it('Should render H4 component without error', () => {
    const component = setupShallow({ children: 'تست', level: 4 });
    const H4 = findByTestAtrr(component, 'h4-tag');
    expect(H4.length).toBe(1);
  });

  it('Should render small H4 component without error', () => {
    const H4 = setupMount({ children: 'تست', level: 4, size: 'sm' });
    expect(H4).toHaveStyleRule('font-size', `${theme.h4SizeSmall}px`);
    expect(H4).toHaveStyleRule('line-height', `${theme.h4SizeSmall * theme.lineHeightRatio}px`);
    expect(H4).toHaveStyleRule('color', theme.defaultColor);
    H4.unmount();
  });

  it('Should render H5 component without error', () => {
    const component = setupShallow({ children: 'تست', level: 5 });
    const H5 = findByTestAtrr(component, 'h5-tag');
    expect(H5.length).toBe(1);
  });

  it('Should render small H5 component without error', () => {
    const H5 = setupMount({ children: 'تست', level: 5, size: 'sm' });
    expect(H5).toHaveStyleRule('font-size', `${theme.h5SizeSmall}px`);
    expect(H5).toHaveStyleRule('line-height', `${theme.h5SizeSmall * theme.lineHeightRatio}px`);
    expect(H5).toHaveStyleRule('color', theme.defaultColor);
    H5.unmount();
  });

  it('Should render H6 component without error', () => {
    const component = setupShallow({ children: 'تست', level: 6 });
    const H6 = findByTestAtrr(component, 'h6-tag');
    expect(H6.length).toBe(1);
  });

  it('Should render medium H6 component without error', () => {
    const H6 = setupMount({ children: 'تست', level: 6, size: 'md' });
    expect(H6).toHaveStyleRule('font-size', `${theme.h6SizeMedium}px`);
    expect(H6).toHaveStyleRule('line-height', `${theme.h6SizeMedium * theme.lineHeightRatio}px`);
    expect(H6).toHaveStyleRule('color', theme.defaultColor);
    H6.unmount();
  });


  it('Should not render H1 when children prop is undefined', () => {
    const component = setupShallow({ level: 1 });
    const H1 = findByTestAtrr(component, 'h1-tag');
    expect(H1.length).toBe(0);
  });

  it('Level prop should be between 1 and 5', () => {
    const H1 = setupMount({ children: 'تست', level: 1 });
    expect(H1.prop('level')).toBeGreaterThanOrEqual(1);
    expect(H1.prop('level')).toBeLessThanOrEqual(5);
    H1.unmount();
  });

  it('Should render H1 when level prop is undefined', () => {
    const component = setupShallow({ children: 'تست' });
    const H1 = findByTestAtrr(component, 'h1-tag');
    expect(H1.length).toBe(1);
  });
});
