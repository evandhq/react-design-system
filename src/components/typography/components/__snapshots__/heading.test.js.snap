// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component Should render H1 component without error 1`] = `
<styled.h1
  data-test="h1-tag"
>
  تست
</styled.h1>
`;

exports[`Heading component Should render H2 component without error 1`] = `
<styled.h2
  data-test="h2-tag"
>
  تست
</styled.h2>
`;

exports[`Heading component Should render H3 component without error 1`] = `
<styled.h3
  data-test="h3-tag"
>
  تست
</styled.h3>
`;

exports[`Heading component Should render H4 component without error 1`] = `
<styled.h4
  data-test="h4-tag"
>
  تست
</styled.h4>
`;

exports[`Heading component Should render H5 component without error 1`] = `
<styled.h5
  data-test="h5-tag"
>
  تست
</styled.h5>
`;
